### PR TITLE
Improve ``PresenterLifeCycleLinker`` public API.

### LINKED ISSUE #51: Improve ``PresenterLifeCycleLinker`` public API.
Now we have changed the ``PresenterLifeCycleLinker`` visibility to public. We should review the public API to improve its usage.

### PR DESCRIPTION
**Pull request by pedrovgs**
Fix #51. Now that we have accepted a pull request (#49) changing the visibility of the class ``PresenterLifeCycleLinker`` from default to public I have decided to improve this class usage. This is the list of improvements you will find in this pull request.

* Simplify the class usage reducing the number of methods needed to link the component lifecycle with the presenter lifecycle.
* Add javadoc.
* Add some guards related to the input params.
* Improve test coverage.
* Improve the project README to show how to use Rosie without extending from ``RosieActivity`` or ``RosieFragment``.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,8 @@ public class SampleApplication extends RosieApplication {
 }
 ```
 
+**Extend from ``RosieApplication`` is needed to be able to easily use the configuration Rosie provides you related to Dependency Injection. If you do not want to use Dependency Injection in your project, you do not need to extend from ``RosieApplication``.**
```
Comment (Serchinastico) [inline README.md]: Use Extend**ing** here and below, or just reword it to `Is needed to extend from... to be able to...`/`Is not mandatory to extend from...`

Comment (pedrovgs) [inline README.md]: Fixed here and in the other comment I've added :)

File: rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java [hunk 2]
```diff
@@ -78,13 +87,63 @@ public void pausePresenters() {
...
+    }
     presenters.add(presenter);
   }
+
+  /**
+   * Updates the view instance associated to all the already registered presenters.
+   *
+   * @param view to be assigned to the presenters.
+   */
+  public void setView(RosiePresenter.View view) {
```
Comment (Serchinastico) [inline rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java]: I can't think of any scenario where I'd update the view and not send the presenter an initialize/update message. Do we need to make this method public?

File: rosie/src/test/java/com/karumi/rosie/view/PresenterLifeCycleLinkerTest.java [hunk 3]
```diff
@@ -68,26 +82,69 @@
...
-    presenterLifeCycleLinker.updatePresenters();
+    presenterLifeCycleLinker.updatePresenters(anyView);
     presenterLifeCycleLinker.pausePresenters();
 
-    verify(anyPresenter1).resetView();
+    assertNotEquals(anyView, presenter.getView());
+  }
+
+  @Test(expected = PresenterNotAccessibleException.class)
+  public void shouldThrowExceptionIfThereAnnotatedPresenterVisibilityIsPrivate() {
```
Comment (Serchinastico) [inline rosie/src/test/java/com/karumi/rosie/view/PresenterLifeCycleLinkerTest.java]: shouldThrowExceptionIf**TheAnnotated**...

**Comment by Serchinastico** [comment]
LGTM!

File: rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java (hunk 2 above)
Comment (pedrovgs) [inline rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java]: We can register a presenter manually and that's why the visibility of this method has to be public. As we can register a presenter which view has not been initialized we need to keep this method to update the view instance if needed. On the other hand, imagine that the framework user wants to change the view without invoke any lifecycle method, we need to keep this method as public.

File: rosie/src/test/java/com/karumi/rosie/view/PresenterLifeCycleLinkerTest.java (hunk 3 above)
Comment (pedrovgs) [inline rosie/src/test/java/com/karumi/rosie/view/PresenterLifeCycleLinkerTest.java]: Fixed!

**Comment by pedrovgs** [comment]
I'll wait until travis pass the build and I'll merge this PR. Thanks!!

File: rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java (hunk 2 above)
Comment (Serchinastico) [inline rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java]: Yup, I understand, I just couldn't think of any case where I'd want to change the view mid-lifecycle but I'm ok if someone uses it
